### PR TITLE
ARROW-3659: [CI] Fix Travis matrix entry 2 documentation to use gcc

### PR DESCRIPTION
**Pull request by pcmoritz**
This is a little subtle and is needed because of the order in which Travis evaluates the command line variable definitions (see the JIRA issue).

I also needed to remove a warning due to this: https://github.com/cython/cython/issues/2269
We might be able to bring the warning back once we can depend on cython with https://github.com/cython/cython/pull/2669

### CODE REVIEW

**Comment by pcmoritz** [comment]
Can somebody help looking into what is going on with the code coverage? I'm not sure how it is supposed to work.

**Comment by pcmoritz** [comment]
Ah ok, thanks!

**Comment by pitrou** [comment]
I don't think we care much about the compiler here, so I'd just rename the CI entry to reflect that it uses gcc, not clang (there are other entries using clang anyway). What do you think?

**Comment by pcmoritz** [comment]
Ok, I replaced it with gcc (4.9 actually to make it compatible with gandiva for https://github.com/apache/arrow/pull/2822) and it should be ready to merge.

File: .travis.yml [hunk 1]
```diff
@@ -80,8 +80,7 @@ matrix:
     - ARROW_TRAVIS_JAVA_BUILD_ONLY=1
     # ARROW-2999 Benchmarks are disabled in Travis CI for the time being
     # - ARROW_TRAVIS_PYTHON_BENCHMARKS=1
-    - CC="clang-6.0"
-    - CXX="clang++-6.0"
+    - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
```
Comment (pitrou) [inline .travis.yml]: Why MATRIX_EVAL?

Comment (pcmoritz) [reply .travis.yml]: because CC and CXX get exported before Travis re-exports them according to the build matrix (see the JIRA issue printout).

Comment (pcmoritz) [reply .travis.yml]: (the other build matrix entries do it the same way)

Comment (pitrou) [reply .travis.yml]: Uh, I see, thanks.

**Review by pitrou — APPROVED** [review]
+1

File: .travis.yml (hunk 1 above)
Comment (kou) [reply .travis.yml]: FYI: We can find the `MATRIX_EVAL` technique in Travis CI document: https://docs.travis-ci.com/user/languages/c/#gcc-on-linux

**Review by kou — APPROVED** [review]
CI is green.
I'll merge this.

**Pull request closed without merging**